src/storage-client/src/controller.rs

```diff
@@ -2268,6 +2277,72 @@ where
                     }
                 }
             }
+            Some(StorageResponse::StatusUpdates(updates)) => {
```

Comment:
we probably want at least some of this branch broken into a function (or method)

Reply:
Done!